### PR TITLE
Use correct clip path method for RoundRectangle

### LINKED ISSUE #9129: Clipping an Image with a Border has a gap on Windows
### Description

XAML is

```
    <Grid BackgroundColor="Blue">
        <Grid.RowDefinitions>
            <!-- label, search -->
            <RowDefinition Height="Auto" />
        </Grid.RowDefinitions>
        <HorizontalStackLayout Grid.Row="0" Margin="20">
            <!-- meal all -->
            <Border Stroke="LightGray" StrokeThickness="3">
                <Border.StrokeShape>
                    <RoundRectangle CornerRadius="20" />
                </Border.StrokeShape>
                <VerticalStackLayout BackgroundColor="#f0f0f0">
                    <Image Aspect="AspectFill" HeightRequest="100" Source="meal_all.jpg" WidthRequest="100" />
                    <Label HorizontalTextAlignment="Center" Margin="0,5" Text="All" />
                </VerticalStackLayout>
            </Border>
            <!-- meal noodle -->
            <Border Stroke="LightGray" StrokeThickness="3">
                <Border.StrokeShape>
                    <RoundRectangle CornerRadius="20" />
                </Border.StrokeShape>
                <Grid BackgroundColor="#f0f0f0">
                    <Grid.RowDefinitions>
                        <RowDefinition Height="Auto" />
                        <RowDefinition Height="Auto" />
                    </Grid.RowDefinitions>
                    <Image Grid.Row="0" Aspect="AspectFill" HeightRequest="100" Source="meal_noodle.jpg" WidthRequest="100" />
                    <Label Grid.Row="1" HorizontalTextAlignment="Center" Margin="0,5" Text="Noodle" />
                </Grid>
            </Border>
            <!-- meal rice -->
            <Border Stroke="LightGray" StrokeThickness="3">
                <Border.StrokeShape>
                    <RoundRectangle CornerRadius="20" />
                </Border.StrokeShape>
                <VerticalStackLayout BackgroundColor="#f0f0f0">
                    <Image Aspect="AspectFill" HeightRequest="100" Source="meal_rice.jpg" WidthRequest="100" />
                    <Label HorizontalTextAlignment="Center" Margin="0,5" Text="Rice" />
                </VerticalStackLayout>
            </Border>
        </HorizontalStackLayout>
    </Grid>
```

Android API 30 is okay

image

Window 10 has gap

image

### Public API Changes

.

### Intended Use-Case

please reduce gap within Border

### PR DESCRIPTION
**Pull request by hartez**
### Description of Change

`RoundRectangle` as a border shape needs to use the `InnerPathForBounds()` method for clipping. 

Making this a draft, it needs device tests.

### Issues Fixed

Fixes #9129

### CODE REVIEW

**Comment by jsuarezruiz** [comment]
Added the device tests from https://github.com/dotnet/maui/pull/14403

File: src/Core/src/Platform/Windows/ContentPanel.cs [hunk 1]
```diff
@@ -104,27 +118,38 @@ void UpdateContent()
...
-
 			var visual = ElementCompositionPreview.GetElementVisual(Content);
 			var compositor = visual.Compositor;
 
 			var pathSize = new Graphics.Rect(0, 0, width, height);
-			var clipPath = clipGeometry.PathForBounds(pathSize);
+
+			PathF? clipPath;
+
+			if (clipGeometry is IRoundRectangle roundedRectangle)
```
Comment (mattleibow) [inline src/Core/src/Platform/Windows/ContentPanel.cs:141]: Why just the round rects? What makes this special? Will we soon see other types being added?

Comment (hartez) [reply src/Core/src/Platform/Windows/ContentPanel.cs:141]: No idea, not my design. I was just making it work the way it worked on the other two platforms.

**Comment by hartez** [comment]
Pretty sure we no longer need this, the target issue has been closed by another PR: https://github.com/dotnet/maui/issues/9129

**Pull request closed without merging**